Sources/LoopUITestingKit/Helpers/Utils.swift

```diff
@@ -190,4 +190,28 @@ extension XCUIApplication {
             maxAttempts -= 1
         }
     }
+    
+    public func waitForExpectedState(
+        expectedState: @escaping () -> Bool,
+        passAfterSeconds: Int = 0,
+        failAfterSeconds: Int = 5,
+        testInfo: XCTestCase
+    ) -> Bool {
+        let expectation = XCTestExpectation(description: "End timer if \(expectedState()) is true")
+        var counter = 0
+        var actualMatch: Bool = false
+        var timer: Timer?
+        timer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { _ in
```

Comment:
```suggestion
        var timer: Timer?
        
        timer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { _ in
```

I would add space between const and vars definition and the code.

Overall the func waitForExpectedState looks good. I was thinking about some changes but it seems that you used the most suitable libraries to define it :) So just few format comments.